src/Simplic.Studio.Tenant.UI/OrganizationViewModel.cs

```diff
@@ -0,0 +1,264 @@
+﻿using Simplic.TenantSystem;
+using Simplic.UI.MVC;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace Simplic.Studio.TenantSystem.UI
+{
+    /// <summary>
+    /// OrganizationViewModel to handle tenant data.
+    /// </summary>
+    public class OrganizationViewModel : ViewModelBase
+    {
+        private IOrganizationService organizationService = CommonServiceLocator.ServiceLocator.Current.GetInstance<IOrganizationService>();
+        private ObservableCollection<Organization> organizations;
+        public int subOrganizationCount;
+        public IList<Guid> subOrganizations;
+        public Guid? cloudOrganizationId;
+        private Organization model;
+        private bool isCreate;
+
+        /// <summary>
+        /// Default constructor for the view model
+        /// </summary>
+        public OrganizationViewModel()
+        {
+            Organizations = new ObservableCollection<Organization>();
+
+            foreach (var organization in organizationService.GetAll())
+            {
+                Organizations.Add(organization);
+            }
+
+            this.IsCreate = true;
+        }
+
+        /// <summary>
+        /// Constructor for the view model
+        /// </summary>
+        /// <param name="id">Tenant ID</param>
+        /// <param name="name">Tenant name</param>
+        /// <param name="matchCode">Tenant match code</param>
+        /// <param name="subOrganizationCount">Tenant suborganizations count</param>
+        /// <param name="subOrganizations">Tenant suborganizations IDs collection</param>
+        /// <param name="cloudOrganizationId">Tenant cloud organization id</param>
+        /// <param name="isActive">Tenant active flag</param>
+        /// <param name="isGroup">Tenant group flag</param>
+        public OrganizationViewModel(Guid id, string name, string matchCode, int subOrganizationCount,
+            IList<Guid> subOrganizations, Guid? cloudOrganizationId, bool isActive, bool isGroup) : this()
+        {
+            if (this.Model == null)
+            {
+                this.model = new Organization
+                {
+                    Id = id,
+                    Name = name,
+                    MatchCode = matchCode,
+                    IsActive = isActive,
+                    SubOrganizations = subOrganizations,
+                    CloudOrganizationId = cloudOrganizationId
+                };
+
+                IsCreate = true;
+            }
+
+        }
+
+        /// <summary>
+        /// Copy constructor
+        /// </summary>
+        /// <param name="org"></param>
+        public OrganizationViewModel(Organization org) : this(org.Id, org.Name, org.MatchCode, org.SubOrganizationCount,
+            org.SubOrganizations, org.CloudOrganizationId, org.IsActive, org.IsGroup)
```

Comment:
calling the other constructor here does nothing but slowing down the instance initialization